### PR TITLE
[WIN32SS][NTUSER] Check window alive before setting focus

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-14945](https://jira.reactos.org/browse/CORE-14945)

### CODE REVIEW

File: win32ss/user/ntuser/winpos.c [hunk 1]
```diff
@@ -2713,8 +2713,12 @@ co_WinPosShowWindow(PWND Wnd, INT Cmd)
       {
          Parent = Wnd->spwndParent;
          if (UserIsDesktopWindow(Wnd->spwndParent))
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/winpos.c]: May as well use `Parent` here too.

Comment (katahiromz) [reply win32ss/user/ntuser/winpos.c]: OK.

**Comment by katahiromz** [comment]
8aeb6a920ac68474a5118b778e7162f53ab17d4e
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24600
https://build.reactos.org/builders/Test%20VBox/builds/11044
https://build.reactos.org/builders/Test%20KVM/builds/23460

9b335e599fc16568e30719de3fbeacbad1bdfaee
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24601
https://build.reactos.org/builders/Test%20VBox/builds/11045

**Comment by SergeGautherie** [comment]
> Analyze these results

Quick glance at TestMan summaries: nothing obviously odd.

**

Only KVM 23461 log has 1 of
```
err:(dll/win32/shell32/COpenWithMenu.cpp:138) Failed to add app zipfldr.dll
(win32ss/user/ntuser/vis.c:50) err: ATM the Current Window or Parent is dead! BC65A8A8
OpenAs_RunDLL.cpp:125: Test failed: Line 64: error expected 1447, was 0
```
I can't say whether it is a regression, random or expected.

**

But what the issue was really about is **Test KVM AHK**...

**Comment by katahiromz** [comment]
https://build.reactos.org/builders/Test%20KVM%20AHK/builds/8380

File: win32ss/user/ntuser/winpos.c [hunk 2]
```diff
@@ -2711,10 +2711,13 @@ co_WinPosShowWindow(PWND Wnd, INT Cmd)
       /* Revert focus to parent */
       if (Wnd == pti->MessageQueue->spwndFocus)
       {
+         USER_REFERENCE_ENTRY Ref;
          Parent = Wnd->spwndParent;
-         if (UserIsDesktopWindow(Wnd->spwndParent))
-             Parent = 0;
+         if (UserIsDesktopWindow(Parent))
+             Parent = NULL;
+         if (Parent) UserRefObjectCo(Parent, &Ref);
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/winpos.c]: Style: I don't like 1-line if(), but maybe that is just me?

Comment (katahiromz) [reply win32ss/user/ntuser/winpos.c]: OK. Fixed.

File: win32ss/user/ntuser/winpos.c [hunk 3]
```diff
@@ -2715,9 +2715,15 @@ co_WinPosShowWindow(PWND Wnd, INT Cmd)
          Parent = Wnd->spwndParent;
          if (UserIsDesktopWindow(Parent))
              Parent = NULL;
-         if (Parent) UserRefObjectCo(Parent, &Ref);
+         if (Parent)
+         {
+            UserRefObjectCo(Parent, &Ref);
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/winpos.c]: Indentation looks to miss 1 space...
If you add braces here, then as well add them to previous if() too.

Comment (katahiromz) [reply win32ss/user/ntuser/winpos.c]: OK.

**Comment by SergeGautherie** [comment]
> https://build.reactos.org/builders/Test%20KVM%20AHK/builds/8380

vDEAD did not crash: encouraging!

**Comment by katahiromz** [comment]
@SergeGautherie So do we have to apply 8380?

**Comment by SergeGautherie** [comment]
> https://reactos.org/testman/compare.php?ids=67578,67691,67692

Assuming a typo, as 67578 seems to have crashed very early.

Try https://reactos.org/testman/compare.php?ids=67689,67691,67692 instead:
identical results without or with vDEAD, which is fine.

> So do we have to apply 8380?

As I understand it, this PR is WIP for the time being and should be further discussed with Jim on Jira.

File: win32ss/user/ntuser/winpos.c [hunk 4]
```diff
@@ -2711,10 +2711,21 @@ co_WinPosShowWindow(PWND Wnd, INT Cmd)
...
-             Parent = 0;
+         if (UserIsDesktopWindow(Parent))
+         {
+             Parent = NULL;
+         }
+         if (Parent)
+         {
+             UserRefObjectCo(Parent, &Ref);
+         }
          co_UserSetFocus(Parent);
```
Comment (HBelusca) [inline win32ss/user/ntuser/winpos.c]: But this one is run even if Parent is NULL? Is it supposed to work?

**Comment by katahiromz** [comment]
Time up. Closed.

**Pull request closed without merging**